sqlglot/dialects/mysql.py

```diff
@@ -547,6 +547,22 @@ def _parse_primary_key_part(self) -> t.Optional[exp.Expression]:
             self._match_r_paren()
             return self.expression(exp.ColumnPrefix, this=this, expression=expression)
 
+        def _parse_primary_key(
+            self, wrapped_optional: bool = False, in_props: bool = False
+        ) -> exp.PrimaryKeyColumnConstraint | exp.PrimaryKey:
+            if not in_props and self._match_set(
```

Comment:
i'm not sure this approach is quite right. i think you only need to add  in parser.py#_parse_primary_key a _parse_id_var(), this will be optional and extend all languase to have a named primary key

Reply:
even though mysql drops it, we could keep it, so i'd probably extend the expression and generation as well